src/meta/api/src/schema_api_impl.rs

```diff
@@ -1071,6 +1080,197 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         Ok(index_metas)
     }
 
+    // virtual column
+
+    async fn create_virtual_column(
+        &self,
+        req: CreateVirtualColumnReq,
+    ) -> Result<CreateVirtualColumnReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let mut retry = 0;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
+
+            let (_, old_virtual_column_opt): (_, Option<VirtualColumnMeta>) =
+                get_pb_value(self, &req.name_ident).await?;
+
+            // If virtual columns already exist, merge them together
+            let virtual_column_meta = if let Some(old_virtual_column) = old_virtual_column_opt {
+                let mut virtual_columns = req.virtual_columns.clone();
+                virtual_columns.extend(old_virtual_column.virtual_columns.clone());
+
+                let virtual_columns_set: HashSet<String> =
+                    HashSet::from_iter(virtual_columns.into_iter());
+                let virtual_columns = Vec::from_iter(virtual_columns_set.into_iter());
+
+                VirtualColumnMeta {
+                    table_id: req.name_ident.table_id,
+                    virtual_columns,
+                    created_on: old_virtual_column.created_on,
+                    updated_on: Some(Utc::now()),
+                    drop_on: old_virtual_column.drop_on,
+                }
```

Comment:
It looks like `VirtualColumnMeta` should provide methods to operate it, such as
- `VirtualColumnMeta::update_columns(add: HashSet<String>, remove: HashSet<String>)` for update columns and the `updated_on` timestamp.
Then this block of codes will be clear and simple